### PR TITLE
Improve support for embedding GLMakie

### PR DESCRIPTION
**Pull request by JamesWrigley**
# Description

I don't think this fixes any specific issue, but it's related to #607.

TL;DR it refactors GLMakie to be aware of whether or not it owns `Screen.gl_screen`, which makes embedding GLMakie a lot easier since it won't attempt to do GLFW-specific things anymore.

The GLMakie API can now basically be used as-is without having to copy-and-paste parts to avoid doing incompatible things. I tested it with CImGui, but it should also be useful for other packages like QML.jl and GtkMakie.jl (cc @barche and @jwahlstrand). I don't think this is breaking in general since `Screen` isn't public, but it will be breaking for packages like GtkMakie.jl and CImGui.jl because the `Screen` type changed.

Here's how it simplified the CImGui integration: https://github.com/Gnimuc/CImGui.jl/commit/53e6117

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Added an entry in CHANGELOG.md (for new features and breaking changes)
- [x] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

**Comment by jwahlstrand** [comment]
I would happily adapt Gtk4Makie.jl to use this if it's merged. Thanks!

File: GLMakie/src/screen.jl [hunk 1]
```diff
@@ -388,19 +405,28 @@ end
...
+    screen = screen_from_pool(config.debugging; window)
     apply_config!(screen, config; start_renderloop=start_renderloop)
     if !isnothing(resolution)
         resize!(screen, resolution...)
     end
     return screen
 end
 
-set_screen_visibility!(screen::Screen, visible::Bool) = set_screen_visibility!(screen.glscreen, visible)
+function set_screen_visibility!(screen::Screen, visible::Bool)
```
Comment (SimonDanisch) [inline GLMakie/src/screen.jl:421]: Should these error instead of returning?
I Think for Makie to work correctly, these will need to be overloaded for `Screen{T}`, so they should not end up here for screens not owned by makie, no?
Same for the other functions that noop, when not owning!

Comment (JamesWrigley) [reply GLMakie/src/screen.jl:422]: Hmm, I originally thought that it flat-out shouldn't be allowed for Makie to touch the window but I guess that should be left up to the integration. Changed it to an error in e2edca596.

**Comment by SimonDanisch** [comment]
Thank you!
